python/paddle/distributed/auto_parallel/local_map.py

```diff
@@ -0,0 +1,280 @@
+# Copyright (c) 2024 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import functools
+from collections.abc import Sequence
+from typing import TYPE_CHECKING, Callable, Optional, Union
+
+import paddle
+import paddle.distributed as dist
+from paddle.utils import flatten, pack_sequence_as
+
+if TYPE_CHECKING:
+    from paddle.distributed import ProcessMesh
+
+__all__ = ["local_map"]
+
+PlacementType = Optional[Sequence[dist.Placement]]
+InputPlacements = Optional[tuple[PlacementType, ...]]
+OutputPlacements = Union[PlacementType, tuple[PlacementType, ...]]
+
+
+def local_map(
+    func: Callable,
+    out_placements: OutputPlacements,
+    in_placements: InputPlacements | None,
```

Comment:
```suggestion
    in_placements: Optional[tuple[list[dist.Placement], ...]],
```
没必要新创建太多类型命名，与框架中其它模块的使用习惯都不相同，反而增加用户理解成本。

Reply:
Done